ci_scripts/check_for_new_public_apis.rb

```diff
@@ -0,0 +1,50 @@
+def get_modified_swift_files(current_dir)
+  files = `git diff --name-only origin/master...`.split("\n").select { |f| f.downcase.end_with?(".swift") && !f.downcase.include?("test") }
+  files.select { |f| File.exist?(f) }
+end
+
+def get_added_lines_change(file)
+  `git diff origin/master... -- '#{file}'`.split("\n").select do |line|
+    line.start_with?('+') && !line.start_with?('+++')
+  end.map { |line| line.delete_prefix('+').strip }
+end
+
+def check_spi_protection(file)
+  in_spi_protected_block = false
+  block_depth = 0
+  public_items = []
+  added_lines = get_added_lines_change(file)
+  previous_line_spi_annotated = false
+
+
+  File.foreach(file) do |line|
+    line = line.strip
+    opening_braces = line.count('{')
+    closing_braces = line.count('}')
+    
+    block_depth += opening_braces - closing_braces
+    if line =~ /@_spi\(\w+\)\s+(public\s+|open\s+)?(class|struct|enum|protocol)\s+\w+/ && opening_braces > 0
```

Comment:
Is `&& opening_braces > 0` accounting for import statements?  If so, consider explicitly having a pattern on "import", but if it makes it more complicated, ignore this comment.

Reply:
Hmm, I don't quite understand why we should account for import statements. Could you give an example?

Reply:
Sorry -- I'm in agreement that we shouldn't account for import statements, I was just trying to understand the code.  It seems that `opening_braces > 0` implies that this statement is false for looking at import statements, correct?  If so, are there any other cases that this `opening_braces > 0` is for?

If not, I was suggesting to update the code to something like:
```
 if line =~ /@_spi\(\w+\)\s+(public\ .... && not_import_statement
```

Reply:
@wooj-stripe Yeah, we could have nested types or multiple classes in a file. So, the opening_braces > 0 condition is primarily there to ensure we are within the scope of a Swift type definition (i.e., class, struct, enum, protocol, typealias). We need to be within this scope to correctly identify new items that should be annotated with "@_spi" modifier.